src/core/receiver/gnss_flowgraph.cc

```diff
@@ -206,7 +206,7 @@ void GNSSFlowgraph::init()
             udp_addr_vec.erase(std::unique(udp_addr_vec.begin(), udp_addr_vec.end()), udp_addr_vec.end());
 
             // Instantiate monitor object
-            GnssSynchroMonitor_ = gnss_synchro_make_monitor(channels_count_,
+            GnssSynchroMonitor_ = gnss_synchro_make_monitor(channels_count_ + 1,
```

Comment:
Oh. This feels hack-y.

Inferring from your change, you believe the monitor block is spinning, calling `work()`; you modified `forecast()` to require 1 input on a "guard" input that you created here and attached to a counter source. If my inference of your assessment of the issue is correct, it is either a bug in GNU Radio or a (different?) logic error in the `forecast()` method.

Reply:
Yes. I see, that monitor block is spinning calling work, but observables block, connected in parallel with monitor block does not. I've modified the monitor block, taking ideas from observables block source and it started consuming 2..3% of CPU core, as expected.
The same extra channel is added to observables block here https://github.com/gnss-sdr/gnss-sdr/blob/next/src/core/receiver/gnss_block_factory.cc#L331